subprojects/core-api/src/main/java/org/gradle/api/cache/CacheResourceConfiguration.java

```diff
@@ -28,18 +27,10 @@
 @Incubating
 @HasInternalProtocol
 public interface CacheResourceConfiguration {
-    /**
-     * Configures the timestamp before which an unused entry will be removed from the cache.
-     *
-     * See {@link #setRemoveUnusedEntriesAfterDays(int)}.
-     */
-    Property<Long> getRemoveUnusedEntriesOlderThan();
 
     /**
      * Sets the timestamp before which unused entries will be removed to be calculated exactly
      * the given number of days previous to the current time.
-     *
-     * See {@link #getRemoveUnusedEntriesOlderThan()}
      */
     void setRemoveUnusedEntriesAfterDays(int removeUnusedEntriesAfterDays);
```

Comment:
I think something like this could work (keeps timestamp internal and exposes a Property to users):

```
// public API
Property<Duration> getEligibleForDeletionAfter()

// internal API
Property<Long> getRemoveUnusedEntriesOlderThan()

getRemoveUnusedEntriesOlderThan().convention(getEligibleForDeletionAfter().map({ duration -> System.currentTimeMillis() - duration.toMillis() }));
```